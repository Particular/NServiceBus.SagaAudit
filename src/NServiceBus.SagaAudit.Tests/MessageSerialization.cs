namespace NServiceBus.SagaAudit.Tests
{
    using System;
    using System.Runtime.CompilerServices;
    using NUnit.Framework;
    using ServiceControl.EndpointPlugin.Messages.SagaState;

    [TestFixture]
    public class MessageSerialization
    {
        [Test]
        [MethodImpl(MethodImplOptions.NoInlining)]
        public void SagaUpdated_serializes_correctly()
        {
            var entity = new SagaUpdatedMessage
            {
                SagaId = Guid.Empty,
                SagaState = "SagaState",
                Endpoint = "Endpoint",
                FinishTime = new DateTime(2017, 10, 30, 9, 22, 17, DateTimeKind.Utc),
                Initiator = new SagaChangeInitiator
                {
                    InitiatingMessageId = "InitiatingMessageId",
                    Intent = "intent",
                    IsSagaTimeoutMessage = true,
                    MessageType = "MessageType",
                    OriginatingEndpoint = "OriginatingEndpoint",
                    OriginatingMachine = "OriginatingMachine",
                    TimeSent = new DateTime(2017, 10, 30, 9, 22, 17, DateTimeKind.Utc)
                },
                IsCompleted = true,
                IsNew = true,
                ResultingMessages = { new SagaChangeOutput
                {
                    Destination = "Destination",
                    MessageType = "MessageType",
                    DeliveryAt = new DateTime(2017, 10, 30, 9, 22, 17, DateTimeKind.Utc),
                    DeliveryDelay = TimeSpan.FromSeconds(4000),
                    Intent = "Intent",
                    ResultingMessageId = "ResultingMessageId",
                    TimeSent = new DateTime(2017, 10, 30, 9, 22, 17, DateTimeKind.Utc)
                }},
                SagaType = "SagaType",
                StartTime = new DateTime(2017, 10, 30, 9, 22, 17, DateTimeKind.Utc)
            };
            var serialized = SimpleJson.SimpleJson.SerializeObject(entity, new MessageSerializationStrategy());
            TestApprover.Verify(serialized);
        }
        
    }
}