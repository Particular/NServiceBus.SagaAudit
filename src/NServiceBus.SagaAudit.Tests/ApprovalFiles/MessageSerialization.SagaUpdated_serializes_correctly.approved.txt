{"SagaState":"SagaState","SagaId":"00000000-0000-0000-0000-000000000000","Initiator":{"InitiatingMessageId":"InitiatingMessageId","MessageType":"MessageType","IsSagaTimeoutMessage":true,"TimeSent":"2017-10-30T09:22:17Z","OriginatingMachine":"OriginatingMachine","OriginatingEndpoint":"OriginatingEndpoint","Intent":"intent"},"ResultingMessages":[{"MessageType":"MessageType","TimeSent":"2017-10-30T09:22:17Z","DeliveryAt":"2017-10-30T09:22:17Z","DeliveryDelay":"01:06:40","Destination":"Destination","ResultingMessageId":"ResultingMessageId","Intent":"Intent"}],"Endpoint":"Endpoint","IsNew":true,"IsCompleted":true,"StartTime":"2017-10-30T09:22:17Z","FinishTime":"2017-10-30T09:22:17Z","SagaType":"SagaType"}